js/web/lib/wasm/jsep/webgpu/ops/common.ts

```diff
@@ -568,7 +568,7 @@ const createIndicesHelper =
 
       const impl = () => {
         const impls = [];
-        if (!useUniform) {
+        if (!useUniform && !vectorize) {
```

Comment:
It's weird to add `vectorize` to common part. `useUniform` should be the only indicator to decide whether use uniform or constant value.

Reply:
Actually these two are independent. Without this vectorize flag, as you have already noticed there are unnecessary variables declare in the shader code.
```
const aData_shape = vec3<u32>(3,4,5);
const aData_strides = vec3<u32>(20,5,1);
const bData_shape = vec3<u32>(3,4,5);
const bData_strides = vec3<u32>(20,5,1);
const outputData_shape = vec3<u32>(3,4,5);
const outputData_strides = vec3<u32>(20,5,1);
```
In this case because we are vectorizing we don't need to set useUniform which would unnecessarily add variables in the Uniform structure.

Reply:
When we don't need to use shape/stride uniforms because we vectorized the input/output tensors then we still don't need to create the variables above. I am using the vectorize flag to prevent generating these variables.

Reply:
useUniform flag specifies wehere the shape and struct variables go, either in to Uniform structure or as separate variable. vectorize flag specifies whether the shape and stride are required or not.